runtime/gateway.go

```diff
@@ -407,16 +420,21 @@ func (gateway *Gateway) setupLogger(config *StaticConfig) error {
 	}
 
 	prodCore := zapcore.NewCore(
-		zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
+		logEncoder,
 		output,
-		zap.InfoLevel,
+		logLevel,
 	)
 	zapLogger := zap.New(
 		NewInstrumentedZapCore(
 			prodCore, gateway.AllHostScope,
 		),
 	)
 
+	gateway.logEncoder = logEncoder
+	if gateway.logWriter == nil {
+		gateway.logWriter = output
```

Comment:
need to always assign this to `output`

In fact; let's not re-use the `logWriter` field and just add a new top level field for `gateway.logWriteSyncer` or something

Reply:
Done. Curious why do you think it's better to not reuse `logWriter`?